documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,473 @@
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /metrics/{uid}:
+    get:
+      tags:
+      - "metrics"
+      operationId: "getmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      - "application/json"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      tags:
+      - "metrics"
+      operationId: "postmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: "body"
+        schema:
+          type: "array"
+          items:
+            $ref: "#/definitions/EventPipeProvider"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+
+# Descriptions of common components
+definitions:
+  # Prometheus line exposition format
+  Metrics:
+    type: string
+    example: "http_requests_total{method='post',code='200'} 1027 1395066363000\nhttp_requests_total{method='post',code='400'}    3 1395066363000"
+  Process:
+    type: object
+    properties:
+      uid:
+        type: string
+      name:
+        type: string
+      commandLine:
+        type: string
+    required:
+      - uid
+  LogFilter:
+    type: object
+    properties:
+      provider:
+        type: string
+      logLevel:
+        type: string
+        enum:
+        - "Trace"
+        - "Debug"
+        - "Information"
+        - "Warning"
+        - "Error"
+        - "Critical"
+  # RFC 7807 Problem details
+  Error:
+    type: object
+    properties:
+      code:
+        type: string
+      title:
+        type: string
+      status:
+        type: integer
+      detail:
+        type: string
+      instance:
+        type: string
+    required:
+      - code
+      - title
+      - status
+      - detail
+      - instance
+  EventPipeConfiguration:
+    type: object
+    properties:
+      providers:
+        type: array
+        items:
+          $ref: "#/definitions/EventPipeProvider"
+      requestRundown:
+        type: boolean
+        default: true
+      circularBufferInMB:
+        type: number
+        default: 256
+  EventPipeProvider:
+    type: object
+    properties:
+      name:
+        type: string
+      keywords:
```

Comment:
Is this meant to be a 64 bit hex number encoded as text? For example the string: "0x1234567890abcdef"

Reply:
I propose, we can just be smart here. "0x" prefix implies it's encoded as a string. If it's a number, that's fine too.

I'm not sure how to represent that in a swagger 🤔

Reply:
I'm fine if we aren't sure how to represent it in swagger as long as we understand what we are specifying. String is probably more reliable than number because there is no risk of the parser trying to convert it to a double internally and losing precision.

> "0x" prefix implies it's encoded as a string

In Json strings will have quotes around them and numbers won't so I don't think there is any ambiguity in that respect? I added it because if you have a string it could be ambiguous whether the numbers are interpreted base 10 or base 16. Prefixing with "0x" makes it explicit that the digits are interpreted base 16.